### PR TITLE
fix(budget): apply v2 api spec  & change from CostType to Provider

### PR DESCRIPTION
**Pull request by sulmoJ**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
- apply v2 api spec to budgetCostTypeSelect
    - costType -> providerFilter

- change from CostType to Provider
image

### CODE REVIEW

File: apps/web/src/services/cost-explorer/budget/budget-detail/modules/budget-info/BudgetDetailInfo.vue [hunk 1]
```diff
@@ -61,8 +48,9 @@ const getTargetLabel = (projects: ProjectReferenceMap) => {
 // LOAD REFERENCE STORE
 (async () => {
     await Promise.allSettled([
-        store.dispatch('reference/project/load'),
-        store.dispatch('reference/projectGroup/load'),
+        allReferenceStore.load('project'),
+        allReferenceStore.load('projectGroup'),
+        allReferenceStore.load('plugin'),
```
Comment (yuda110) [inline apps/web/src/services/cost-explorer/budget/budget-detail/modules/budget-info/BudgetDetailInfo.vue:53]: praise: nice

File: apps/web/src/services/cost-explorer/budget/budget-detail/modules/budget-info/BudgetDetailInfo.vue [hunk 2]
```diff
@@ -4,51 +4,38 @@ import { computed, reactive } from 'vue';
...
-    projects: computed<ProjectReferenceMap>(() => store.getters['reference/projectItems']),
-    projectGroups: computed<ProjectGroupReferenceMap>(() => store.getters['reference/projectGroupItems']),
+    projects: computed<ProjectReferenceMap>(() => allReferenceStore.getters.project),
+    projectGroups: computed<ProjectGroupReferenceMap>(() => allReferenceStore.getters.projectGroup),
+    providers: computed<ProviderReferenceMap>(() => allReferenceStore.getters.provider),
     budgetData: computed<BudgetModel|null>(() => budgetPageState.budgetData),
-    costTypeKey: computed(() => {
-        if (!budgetPageState.budgetData || !budgetPageState.budgetData?.cost_types) return '';
-        return getKeyOfCostType(budgetPageState.budgetData.cost_types);
+    processedProviderValue: computed(() => {
```
Comment (yuda110) [inline apps/web/src/services/cost-explorer/budget/budget-detail/modules/budget-info/BudgetDetailInfo.vue:34]: quibble: please set type here! 🥹

**Review by yuda110 — APPROVED** [review]
LGTM!!!

**Review by piggggggggy — APPROVED** [review]
It's so clear! LGTM!

**Pull request merged**